platform/ios/src/MGLMapView.mm

```diff
@@ -5634,38 +5669,38 @@ - (void)didUpdateLocationWithUserTrackingAnimated:(BOOL)animated
         if (self.userTrackingState != MGLUserTrackingStateBegan)
         {
             // Keep both the user and the destination in view.
-            [self didUpdateLocationWithTargetAnimated:animated];
+            [self didUpdateLocationWithTargetAnimated:animated completionHandler:completion];
         }
     }
     else if (self.userTrackingState == MGLUserTrackingStatePossible)
     {
         // The first location update is often a great distance away from the
         // current viewport, so fly there to provide additional context.
-        [self didUpdateLocationSignificantlyAnimated:animated];
+        [self didUpdateLocationSignificantlyAnimated:animated completionHandler:completion];
     }
     else if (self.userTrackingState == MGLUserTrackingStateChanged)
     {
         // Subsequent updates get a more subtle animation.
-        [self didUpdateLocationIncrementallyAnimated:animated];
+        [self didUpdateLocationIncrementallyAnimated:animated completionHandler:completion];
     }
     [self unrotateIfNeededAnimated:YES];
```

Comment:
This method call can result in a snap-to-north animation. Would it be surprising if the animation could occur after the completion handler executes? Accounting for this edge case may complicate the code somewhat.

Reply:
Preferably at the very last chained animation but I don't have a strong opinion against that. However, unrotate is private so it shouldn't complex to hook up too much.